src/main/scala/com/lightbend/kafka/scala/streams/ImplicitConversions.scala

```diff
@@ -35,5 +36,28 @@ object ImplicitConversions {
 
   implicit def Tuple2ToKeyValue[K, V](tuple: (K, V)): KeyValue[K, V] = new KeyValue(tuple._1, tuple._2)
 
-}
+  // technique for optional implicits adopted from 
+  // http://missingfaktor.blogspot.in/2013/12/optional-implicit-trick-in-scala.html
+
+  case class Perhaps[E](value: Option[E]) {
```

Comment:
It probably should be made `final` and extend `AnyVal` to reduce memory allocation and improve inlining.

Reply:
Yeah true, but in the latest release we have got rid of this abstraction altogether.